### ISSUE TITLE
Star Rating as Required Field Not Working

### ISSUE THREAD

**Issue by kairey64** [CalderaWP/Caldera-Forms#1710]
## What Is The Unexpected Beahviour?

Can you tell me if this is a known bug that will be fixed or an issue with something my side. I have disabled caches, tried on other hosting etc. etc. and cannot get rid of the problem.

Basically star ratings get skipped even when they are a required field, I have a customer survey form with a few pages and you can bypass whole pages with star rating fields that are set to required. Any text fields radio buttons etc obey the required rule.

I am running the latest of everything WP, plugins, PHP7, have spoken to the cache plugin people, my hosting company and cannot find any issues our side. The strange this is different browsers and devices seem to behave differently! Just checked one on Chrome and a page with 3 star rating fields actually highlighted one as required, and it just skipped the other 2.

This is a live site so be careful with testing as client receives responses but see https://mahikibeach.com/forms/customer-survey/ this same template is across 5 other sites and all display the same.

KEVIN

**Comment by kairey64** [comment]
Strange I updated to 1.5.3 and this issue seemed to resolved on first time form use. Then I informed the client it was fixed and now notice the issue is back again. I not that this is now scheduled for 1.5.4 but is there a fix I can push through now?

**Comment by kairey64** [comment]
NO! Now put back to 1.5.5 this is killing me?

Can I fix this manually with your help, its a massive issue across feedback forms?

**Comment by Shelob9** [comment]
@kairey64  Thanks for using Caldera Forms. Sorry about this issue. It got reprioritized off of 1.5.4 as that was a critical update for some users and went out ahead of time.

If you want to put in a pull request that would be appreciated. I think the issue is that 0 is a valid value for a text input and the star rating field is a fancy way to set a text field that is hidden using inline CSS. A custom Parsely validator is probably needed here.

**Comment by Shelob9** [comment]
Close via c263fe4 and  7ceb423

**Comment by Shelob9** [comment]
Technically a duplicate of #1634

**Comment by korinektomas** [comment]
Having the same problems with this (version 1.7.6, latest versions of everything, PHP 7.1, no cache)... 

If I set star rating as required it acts completely weird. Sometimes it validates it (show error of field and scroll to the element), but sometimes the processing of the whole form justs "stops" without showing anything. There were also cases, the error message was displayed, but the form did not scroll to the problematic element... I tried to find some pattern in this, but I was not successful :/ 

Is there something I am missing? Or is there some easy workaround? I tried to validate the hidden score input by Custom Callback, but it seems this JS function is processed only after successful validation of Caldera (which is too late)...

**Comment by korinektomas** [comment]
Ok, so I was testing it again and I realized the buggy behavior. The star rating element validates only when it is hovered. When it keeps untouched before submitting the form, the whole process stops without showing any error. I tried to set default value 0 in admin, but this does not help.
